ni845x_spi.c

```diff
@@ -0,0 +1,501 @@
+                }
+            } else {
+                // serial number parameter specified -> check the resource_handle
+                // and if it is not the device what we are looking for
+                // loop on the available devices
+                do {
+                    if (strtol(serial_number, NULL, 16) == serial_as_number) {
+                        if (ni845x_spi_open_resource(resource_handle, returnHandle) == 0)
+                            ret = 0;
+                        break;
+                    }
+
+                    // serial does not match the requested -> continue with the next one
+                    found_devices_count--;
+                    if (found_devices_count) {
+                        if (ni845xFindDeviceNext(device_find_handle, resource_handle) == 0) {
+                            sscanf(resource_handle, "USB%d::0x%04X::0x%04X::%08X::RAW",
+                                   &usb_bus, &vid, &device_pid, &serial_as_number);
+                        }
+                    }
+                } while (found_devices_count);
+            }
+        }
+    } else {
+        ni845xStatusToString(tmp, ARRAY_SIZE(error_string_buffer), error_string_buffer);
+        msg_perr("ni845xFindDevice failed with: %s (%ld)\n", error_string_buffer, tmp);
+        return -1;
+    }
+
+    tmp = ni845xCloseFindDeviceHandle(device_find_handle);
+    if (tmp) {
+        ni845xStatusToString(tmp, ARRAY_SIZE(error_string_buffer), error_string_buffer);
+        msg_perr("ni845xCloseFindDeviceHandle failed with: %s (%ld)\n", error_string_buffer, tmp);
+        return -1;
+    }
+    return ret;
+}
+
+/**
+ * @brief ni845x_spi_open_resource
+ * @param resource_handle the resource handle returned by the ni845xFindDevice or ni845xFindDeviceNext
+ * @param opened_handle the opened handle from the ni845xOpen
+ * @return the 0 on successful competition, negative error code on failure positive code on warning
+ */
+static int32 ni845x_spi_open_resource(char *resource_handle, uInt32 *opened_handle)
+{
+    int32 ret = ni845xOpen(resource_handle, opened_handle);
+    if (ret < 0) {
+        ni845xStatusToString(ret, ARRAY_SIZE(error_string_buffer), error_string_buffer);
+        msg_perr("ni845xOpen(%s) failed with: %s (%ld)\n", resource_handle, error_string_buffer, ret);
+    } else if (ret > 0) {
+        ni845xStatusToString(ret, ARRAY_SIZE(error_string_buffer), error_string_buffer);
+        msg_pwarn("ni845xOpen(%s) failed with: %s (%ld)\n", resource_handle, error_string_buffer, ret);
+    }
+    return ret;
+}
+
+/**
+ * @brief usb8452_spi_set_io_voltage sets the IO voltage for the USB-8452 devices
+ * @param IOVoltage_mV the desired IO voltage in mVolts
+ * @return 0 on success, negative on error, positive on warning
+ */
+static int usb8452_spi_set_io_voltage(uint16_t IOVoltage_mV)
+{
+    int i = 0;
+    uint8_t selected_voltage_100mV = 0;
+    uint8_t IO_voltage_100mV = 0;
+
+    if (device_pid != USB8452) {
```

Comment:
`if (device_pid == USB8451)` so the check will work if other versions are added to the list.

Reply:
Fixed